gammapy/maps/hpxnd.py

```diff
@@ -204,17 +204,10 @@ def sum_over_axes(self):
         map_out : `~HpxNDMap`
             Summed map.
         """
-        hpx_out = self.geom.to_image()
-        map_out = self.__class__(hpx_out)
-
-        if self.geom.nside.size > 1:
-            vals = self.get_by_idx(self.geom.get_idx(flat=True))
-            map_out.fill_by_coord(self.geom.get_coord(flat=True)[:2], vals)
-        else:
-            axis = tuple(range(self.data.ndim - 1))
-            map_out.data = np.sum(self.data, axis=axis)
-
-        return map_out
+        geom = self.geom.to_image()
+        axis = tuple(range(self.data.ndim - 1))
+        data = np.nansum(self.data, axis=axis)
+        return self.clone(geom=geom, data=data)
```

Comment:
Why did you remove the case for multi resolution map? Does this implementation work for healpix maps with mutliresolution scales?
Again the logic of cloning with a new geom is a bit strange to me. You just keep `unit` and `meta`. You don't gain much in terms of complexity with: `map_out = self.__class__(geom=geom, meta=meta, unit=unit)`